checker/src/org/checkerframework/checker/signedness/SignednessVisitor.java

```diff
@@ -34,6 +31,24 @@ private boolean isMask(Tree node) {
         return kind == Kind.AND || kind == Kind.OR;
     }
 
+    /** @return true iff node is a primitive type cast */
+    private boolean isPrimitiveCast(Tree node) {
+        if (node.getKind() != Kind.TYPE_CAST) {
+            return false;
+        }
+
+        TypeCastTree cast = (TypeCastTree) node;
+        Tree castType = cast.getType();
+        if (castType.getKind() != Kind.ANNOTATED_TYPE) {
```

Comment:
I believe that by the time the visitor traverses the AST all types are wrapped by some annotated type, but it is possible that I'm wrong about this.